util/throttle.cc

```diff
@@ -235,15 +235,20 @@ ThrottleThread(
             }   // else
 
             // change the throttle slowly
+            temp_throttle=gThrottleRate;
```

Comment:
If you're concerned about accessing gThrottleRate with a memory barrier, it seems like you should use GetThrottleWriteRate() instead of the raw variable (in all the places it shows up below). Either that, or call GetThrottleWriteRate() once and then cache that value locally and use the cached value as you adjust temp_throttle.